Dedicated Security Content

ahardin-rh opened the pull request:
Updated Preview Build:
http://file.rdu.redhat.com/~ahardin/04252019/dedicated-security/product_security/

ahardin-rh commented:
@jeremyeder @anjalitelang PTAL at my first pass at bringing the Security content into the Dedicated docs for 3.11. Thanks!

jeremyeder commented:
Looks good. Understand you're restructuring a bit and will have another version shortly.

ahardin-rh commented:
@jeremyeder @anjalitelang Thank you! My rework is now complete. Please review the latest preview build and let me know if any changes are needed:
http://file.rdu.redhat.com/~ahardin/04232019/dedicated-security/product_security/index.html

anjalitelang commented on product_security/recommendations_for_customers/customer_logging_metrics_monitoring.adoc:
```diff
@@ -0,0 +1,20 @@
+[[dedicated-security-logging-metrics-monitoring-customers]]
+= Logging, Metrics, and Monitoring
```
Change to Audit logs

anjalitelang commented on product_security/recommendations_for_customers/customer_logging_metrics_monitoring.adoc:
```diff
@@ -0,0 +1,20 @@
+permission to view. Kibana can be accessed at:
+
+----
+https://logs.[cluster-name].openshift.com
```
Add text: Audit logs are maintained by Red Hat for Compliance purposes and are not shareable with customers at this time.

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+* GitHub OAuth
+* OpenID Connect
+* LDAP
+
```
Add text and link: 
Learn more about Authentication in OpenShift Dedicated at : http://file.rdu.redhat.com/~ahardin/04232019/dedicated-security/architecture/additional_concepts/authentication.html

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+of the underlying infrastructure solution.
+
+[[security-authorization]]
+== Authorization
```
Can we add this text right below the Authorization heading:

Authorization is managed using:
Rules : Sets of permitted verbs on a set of objects. For example, whether something can create pods.
Roles : Collections of rules. Users and groups can be associated with, or bound to, multiple roles at the same time.
Bindings: Associations between users and/or groups with a role.

OpenShift Dedicated evaluates authorizations using the following steps:

1. The identity and the project-scoped action is used to find all bindings that apply to the user or their groups.
2. Bindings are used to locate all the roles that apply.
3. Roles are used to find all the rules that apply.
4. The action is checked against each rule to find a match.
5. If no matching rule is found, the action is then denied by default.

ahardin-rh replied:
Is it okay if I omit the text and just move the link up to the content in xref:../../architecture/additional_concepts/authorization.adoc#architecture-additional-concepts-authorization[Learn
how authorization is managed] instead since that text is pulled from there? I don't want to duplicate content in our doc set.

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+[[security-RBAC]]
+=== Role-based Access Control (RBAC)
+
+* xref:../../architecture/additional_concepts/authorization.adoc#architecture-additional-concepts-authorization[Learn
```
Move this link above right after the text for Authorization

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+== Authorization
+
+[[security-RBAC]]
+=== Role-based Access Control (RBAC)
```
RBAC should be a sub-heading under "Authorization"

ahardin-rh replied:
It is. This is an h3 heading while Authorization is an h2. The TOC reflects this, but I know the heading sizes are very similar and hard to distinguish at times.

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+Managing Role-based Access Control (RBAC)].
+
+[[security-cluster-roles-local-roles]]
+=== Cluster Roles and Local Roles
```
Cluster roles and local roles Should be a sub-heading under RBAC.

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+* xref:../../architecture/additional_concepts/authorization.adoc#architecture-additional-concepts-authorization[Learn
+how authorization is managed].
+
+* xref:../../admin_guide/manage_rbac.adoc#admin-guide-manage-rbac[Learn about
```
Move the reference at the end of RBAC paragraph (Right before we start talking about SCC)

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+
+* xref:../../admin_guide/manage_rbac.adoc#admin-guide-manage-rbac[Learn about
+Managing Role-based Access Control (RBAC)].
+
```
Add text:
There are two levels of RBAC roles and bindings that control authorization:

Cluster RBAC : Roles and bindings that are applicable across all projects. Roles that exist cluster-wide are considered cluster roles. Cluster role bindings can only reference cluster roles.

Local RBAC : Roles and bindings that are scoped to a given project. Roles that exist only in a project are considered local roles. Local role bindings can reference both cluster and local roles.

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+can be performed on a set of resources. {product-title} includes a set of
+default cluster roles that can be bound to users and groups cluster wide or
+locally.
+
```
Add link. 
For more information on the roles supported on the cluster see http://file.rdu.redhat.com/~ahardin/04232019/dedicated-security/architecture/additional_concepts/authorization.html#roles

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+locally.
+
+[[security-scc]]
+=== Security Context Constraints
```
SCC should be a sub-heading under Authorization

ahardin-rh replied:
It is. This is an h3 heading while Authorization is an h2. The TOC reflects this, but I know the heading sizes are very similar and hard to distinguish at times.

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+
+[[security-scc]]
+=== Security Context Constraints
+
```
Add text below SCC heading:
Security context constraints allow administrators to control permissions for pods. As an OpenShift Dedicated cluster administrator, you can list and view details for SCCs, but cannot edit or delete the default SCCs.

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,105 @@
+Security Context Constraints].
+
+[[securing-red-har-personnel-access]]
+==== Red Hat Personnel Access
```
Red Hat Personnel Access should be under "Authorization" header

anjalitelang commented on product_security/red_hat_responsibilities/dedicated_network_security.adoc:
```diff
@@ -0,0 +1,98 @@
+administrators to configure their own isolation policies using NetworkPolicy
+objects. These policies can be fully managed by project owners and
+`dedicated-admin` users to add network-level isolation for the projects.
+
```
Add following below:
Additionally, some dedicated clusters also support the ovs-multitenant plug-in that provides project-level isolation for pods and services. Each project receives a unique Virtual Network ID (VNID) that identifies traffic from pods assigned to the project. By default, Pods from different projects cannot send packets to or receive packets from pods and services of a different project. However, the cluster administrator can join projects and later isolate them using admin commands. For more information on Pod network management see https://docs.openshift.com/dedicated/admin_guide/managing_networking.html

anjalitelang commented on product_security/red_hat_responsibilities/dedicated_network_security.adoc:
```diff
@@ -0,0 +1,98 @@
+* router ELB
+* External master ELB
+* Internal master ELB
+
```
Add link: https://docs.openshift.com/container-platform/3.11/install_config/configuring_aws.html#configuring-a-security-group-aws

anjalitelang commented on product_security/red_hat_responsibilities/dedicated_network_security.adoc:
```diff
@@ -0,0 +1,98 @@
+
+[[network-segregation-security-groups]]
+=== Security Groups
+
```
Add following text:
Security Groups provide security at the protocol and port access level. Each security group works similar to a firewall and has a set of rules that filter traffic coming into and out of a node.

anjalitelang commented on product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc:
```diff
@@ -0,0 +1,75 @@
+:prewrap!:
+
+toc::[]
+
```
Needs a heading called Logging

anjalitelang commented on product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc:
```diff
@@ -0,0 +1,75 @@
+then contacts each node individually and scrapes the metrics for CPU, memory,
+and network usage. It then exports these into Hawkular Metrics.
+
+[[customer-controlled-monitoring]]
```
This is my bad. The Customer controlled monitoring needs to go under Recommendations for customers, with Logging.

anjalitelang commented on product_security/responsibilities_of_cloud_providers/index.adoc:
```diff
@@ -0,0 +1,18 @@
+infrastructure including but not limited to secure hardware, software,
+networking, and secure configuration of services offered on the public cloud. At
+this time, {product-title} runs on the Amazon Web Service cloud and inherits
+the security features of that platform.
```
Add link to AWS security: https://aws.amazon.com/security/

anjalitelang commented on product_security/recommendations_for_customers/restrict_access.adoc:
```diff
@@ -0,0 +1,200 @@
+Service Accounts] for more information.
+
+[[security-service-accounts]]
+== OOTB Application-level Public Key Infrastructure
```
Rename to Service Serving Certificate Secrets - Application level PKI

anjalitelang commented on product_security/recommendations_for_customers/secure_container_lifecycle_management.adoc:
```diff
@@ -0,0 +1,92 @@
+a vulnerability is discovered after an image has been deployed, you want yet
+another layer of security: tools for automated, policy-based deployment. Strong
+security includes automated policies that you can use to manage container
+deployment from a security point of view.
```
At the end of the whole page, maybe we can add this reference link https://www.redhat.com/en/resources/container-security-openshift-cloud-devops-whitepaper#

ahardin-rh replied:
Good idea. I am adding an Additional Resources section with this link

anjalitelang commented on product_security/recommendations_for_customers/restrict_access.adoc:
```diff
@@ -0,0 +1,200 @@
+
+See the
+xref:../../architecture/networking/routes.adoc#architecture-core-concepts-routes[Routes]
+topic for an overview of routes, including route-specific annotations.
```
add after including route-specific annotations - " and IP whitelisting"

ahardin-rh replied:
Adding in the text, but I'll leave it unbolded per our style guidelines

ahardin-rh commented:
@anjalitelang Thanks for your feedback! I addressed your comments. Please let me know if I missed anything. The updated preview build can be reviewed here:
http://file.rdu.redhat.com/~ahardin/04252019/dedicated-security/product_security/

@jeremyeder PTAL as well. Thanks!

ahardin-rh commented:
@anjalitelang Thank YOU! I updated  "Customer-controlled Monitoring" to "Monitoring". Unfortunately, I can't address your second point, as this is our style for all h4 headings. Please let me know if anything else needs addressed.

@patrickstrick @jeremyeder @spurtell @jgouldin @dbaker-rh @rhdedgar Please review this content for the Dedicated 3.11 Security Guide. Thank you!

anjalitelang commented on product_security/recommendations_for_customers/secure_container_lifecycle_management.adoc:
```diff
@@ -0,0 +1,96 @@
+Hat Enterprise Linux Host, which is the underlying immutable operating system
+deployed with {product-title}. Linux features such as Security-Enhanced Linux,
+namespaces, Cgroups, and seccomp are used to secure the container. Containers
+are run with user, without a fixed UID, and not root privileges on the host.
```
Replace with - Container processes run under a UID assigned by the OpenShift Platform, and never with root privileges.

anjalitelang commented:
@patrickstrick @jeremyeder @spurtell @jgouldin  @rhdedgar Please review this content for the Dedicated 3.11 Security Guide. Thank you!

patrickstrick commented on product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc:
```diff
@@ -0,0 +1,67 @@
+and network usage. It then exports these into Hawkular Metrics.
+
+[[security-monitoring-and-testing]]
+== Monitoring and Testing
```
This section confuses me. What do we mean by "monitoring service"? The 2nd sentence then talks about logging. Then, the 3rd sentence says automated testing, config mgmt, etc. This seems like it is talking about three different things.

anjalitelang replied:
Background: I got this section from the Service definition mainly to highlight how we monitor the cluster. We did not want to use Zabbix reference so used the term centrallized logging instead for aggregation of the logs. 
Recommended Change:  
{product-title} clusters come with an integrated Prometheus/Grafana stack for cluster monitoring. Monitoring service containers runs on each application node and scrap data from the nodes which is then fronted on the Grafana dashboard. The aggregated data is further used by Red Hat SRE for services such as pruning, garbage collection, automated testing and configuration management.

patrickstrick commented on product_security/red_hat_responsibilities/compliance.adoc:
```diff
@@ -0,0 +1,21 @@
+
+toc::[]
+
+Compliance of Red Hat {product-title} on AWS against compliance standards such
```
```suggestion
Certificaiton of Red Hat {product-title} on AWS against compliance standards such
```
?

patrickstrick commented on product_security/red_hat_responsibilities/compliance.adoc:
```diff
@@ -0,0 +1,21 @@
+Compliance of Red Hat {product-title} on AWS against compliance standards such
+as SOC, ISO, PCI DSS, and HIPAA is in progress. As a hosted service provider,
+Red Hat has strong processes aligned with NIST 800-123 standards to manage
+information security. Red Hat implements security controls that ascertain
```
I am not sure we should use the language "ascertain" and "ensuring" as those may be legally binding. Check with legal first.

ahardin-rh replied:
@patrickstrick Okay, thanks! I will try to track someone down.

ahardin-rh replied:
@patrickstrick @anjalitelang I opened a ticket with Legal and copied you. The rest of your feedback is now addressed. Thanks!

patrickstrick commented on product_security/recommendations_for_customers/disaster_recovery.adoc:
```diff
@@ -0,0 +1,19 @@
+
+Planning for disaster recovery is always a crucial component of a sound security
+posture. While Red Hat provides backup of project related resources it
+is the responsibility of the customer to backup resources application specific
```
```suggestion
is the responsibility of the customer to backup application specific
```

patrickstrick commented on product_security/recommendations_for_customers/disaster_recovery.adoc:
```diff
@@ -0,0 +1,19 @@
+Planning for disaster recovery is always a crucial component of a sound security
+posture. While Red Hat provides backup of project related resources it
+is the responsibility of the customer to backup resources application specific
+data for example, special export, and import procedures for database systems. In
```
```suggestion
data for example, special export and import procedures for database systems. In
```

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,130 @@
+Tiered access is how Red Hat grants internal personnel limited access to
+{product-title} environments.
+
+* Users can write and submit read-only scripts that allow a subset of
```
We originally had Tier[x] added but removed it as it is too detailed for product docs.

anjalitelang commented on product_security/red_hat_responsibilities/access_control.adoc:
```diff
@@ -0,0 +1,130 @@
+by the Red Hat Customer Experience and Engagement team to provide support for
+customer cases.
+
+* Users granted this level of access are given OpenShift cluster-reader access.
```
We originally had Tier[x] added but removed it as it is too detailed for product docs.

anjalitelang commented on product_security/red_hat_responsibilities/dedicated_network_security.adoc:
```diff
@@ -0,0 +1,105 @@
+== Encryption for Data at Rest
+
+All new persistent volumes (PVs) are encrypted at rest. PVs are backed by block
+storage, which is Read-Write-Once and will be dynamically provisioned and
```
+1

anjalitelang commented on product_security/red_hat_responsibilities/dedicated_network_security.adoc:
```diff
@@ -0,0 +1,105 @@
+2048 bit RSA keys and SHA-256 signatures.
+
+The insecure versions SSL 2.0 and SSL 3.0 are unsupported and not available. The
+{product-title} server and `oc` client only provide TLS 1.2 or newer by default.
```
+1 Patrick.  I think this just got left out in the comments and never got fixed. 
We default when I do openssl s_client -connect api.bu-demo.openshift.com:443, the cert offered is TLS1.2.. but yes I tested with TLS1.1/SSLv3 and TLS1.0 and it connected. Rephrasing to not include SSLv2/v3 reference and adding
The
{product-title} server and `oc` client provide TLS 1.2 or newer by default.

anjalitelang commented on product_security/red_hat_responsibilities/dedicated_network_security.adoc:
```diff
@@ -0,0 +1,105 @@
+PKI is created for each deployment of {product-title}. The internal PKI uses
+2048 bit RSA keys and SHA-256 signatures.
+
+The insecure versions SSL 2.0 and SSL 3.0 are unsupported and not available. The
```
Replace lines 73 to remove SSL reference.
Replace Line 74 with The
{product-title} server and `oc` client provides TLS 1.2 or newer by default.

anjalitelang commented on product_security/red_hat_responsibilities/logging_metrics_monitoring.adoc:
```diff
@@ -0,0 +1,67 @@
+[[security-dedicated-logging]]
+== Logging
+
+{product-title} uses an Elasticsearch Fluentd Kibana (EFK) stack to collect logs
```
+1

ahardin-rh commented:
Per discussion with @patrickstrick and @anjalitelang, we removed a sentence that is currently being reviewed by Red Hat Legal. All other comments from @patrickstrick @anjalitelang and @dbaker-rh are addressed, so I am merging this. If Legal comes back with approval on the Compliance verbiage, we can add it back in via a separate PR.

ahardin-rh commented:
/cherrypick enterprise-3.11
